Support adding sender and reply-to name for sending email

carmenlau opened the pull request:
connect #34

carmenlau committed:
Support sender name and reply-to name in smtp provider, for email verification

carmenlau committed:
Support sender name and reply-to name to welcome and forgot password email

carmenlau committed:
Add test for mailer payload

cheungpat commented on forgot_password/handlers/util/email.py:
```diff
@@ -25,14 +25,25 @@ def __init__(self, **smtp_params):
 
-        if reply_to:
-            headers.append(('Reply-To', reply_to))
+        if reply_to and reply_to[1]:
```
I think you can simply pass `reply_to` to `format_addresses`. `format_addresses` support both single string (an email address) or a tuple (name-email pair), see https://github.com/aspineux/pyzmail/blob/e1a19edc86dbae59b67e83dd8a0ef5ee5f663dbb/pyzmail/pyzmail/generate.py#L93.

cheungpat replied:
This is different from sender because pyzmail takes the second element of `sender` tuple (i.e. `sender[1]`) as a value to the SMTP `MAIL FROM` address.

cheungpat commented on forgot_password/handlers/forgot_password.py:
```diff
@@ -124,7 +125,9 @@ def test_forgot_password_email(email,
                                    sender=None,
-                                   reply_to=None):
+                                   reply_to=None,
+                                   sender_name='',
```
Does the underlying library works if you use `None`, as far as I checked, the underlying library is smart enough to omit the name if the value is false.

Well... just trying to make this consistent with other params. : )

cheungpat commented on forgot_password/handlers/util/email.py:
```diff
@@ -25,14 +25,25 @@ def __init__(self, **smtp_params):
         Send email to user.
+
+        Arguments:
+        sender - (tuple) a tuple of the form ('Name', 'sender@example.com')
```
While the underlying library expects the the sender to be a tuple (just the `sender` has to be a tuple, this doesn’t apply the same to `reply_to`), I think we should check the type of the parameters here, and convert a simple string to a tuple. This helps user of this util as giving a string param is much easier.

carmenlau committed:
Update email util to accept both email and reply to in string format.

- Update email util to accept both email and reply to in string format
- Change sender and reply to name to None as default in email testing
api

carmenlau commented:
Updated and thanks @cheungpat !
I kept `reply_to_tuple and reply_to_tuple[1]` checking as I only want to append the header if there is reply to email. See if the change looks good for you? :)
